src/pages/index.js

```diff
@@ -83,87 +82,109 @@ function handleDeleteIcon(cardId, card) {
     api
       .deleteCard(cardId)
       .then(() => {
-        deleteCardPopup.close();
         card.remove();
       })
       .catch((err) => {
         console.error(err);
+      })
+      .finally(() => {
+        deleteCardPopup.close();
       });
   });
 }
 
 function handleProfileImgFormSubmit(data) {
   api
     .updateProfileImg(data.link)
+    .then(profileImagePopup.editSubmitBtn())
     .then(() => {
       userInfo.setUserProfileImage(data.link);
-      profileImagePopup.close();
     })
-    .catch((error) => {
-      console.error("Error fetching data:", error);
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileImagePopup.resetSubmitBtn();
+      profileImagePopup.close();
     });
 }
 
 function handleCardClick(name, link) {
   previewImagePopup.open(name, link);
 }
-function addLike(cardId, likeCounter) {
+function addLike(cardId, likeCounter, likeBtn) {
```

Comment:
**Could be improved**

To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

        this._likeButton.addEventListener('click', () => {
            this._handleLike(this);
        });


```

And now in `index.js` you can call the methods like this:


```js


function addLike(cardInstance) {
        api.likeCard(cardInstance.getId())
            .then((res) => {
               cardInstance.addLike(res)
            })
            .catch((err) => {
                console.error(err);
            });
}


```

The same thing is with deleting cards